pkg/resend/client.go

```diff
@@ -41,7 +41,7 @@ func (c *Client) SendEmailAsync(ctx context.Context, subject, text, html string,
 		Fn: func(ctx context.Context) error {
 			_, err := c.client.Emails.SendWithContext(ctx, req)
 			if err != nil {
-				return emailerrors.NewEmailSendingFailedError().Wrap(err)
+				return fmt.Errorf("failed to send email via Resend: %w", err)
```

Comment:
Emails are sent asynchronously, and potential errors are only logged, not handled in the service. Therefore, there is no need to have `pkg/emailerrors`.